Testscripts/Linux/channel_change.sh

```diff
@@ -0,0 +1,179 @@
+							cpu_idx=$((cpu_idx+1))
+						else
+							LogErr "Failed to change the cpu id of the channel $rel_id from ${idle_cpus[$cpu_idx]} to 0. Expected 0, but found $_cpu_id"
+						fi
+					fi
+					read -a line
+				done
+				#LogMsg "Found vmbus channel and its target cpus: ${temp[@]}, ${!temp[@]}"
+			else
+				LogErr "Supposed to read Sysfs path, but found ${line[@]}"
+			fi
+			idx=$((idx+1))
+		else
+			LogErr "Supposed to read VMBUS ID line, but found ${line[@]}"
+		fi
+		LogMsg ""
+	done < "$lsvmbus_output_location"
+
+	LogMsg "VMBUS scanning result: ${vmbus_id[@]}"
+	return
+}
+
+function Main() {
+	# Collect the vmbus and cpu id information from the system, and store in /tmp/lsvmbus.output
+	basedir=$(pwd)
+	lsvmbus
+	if [ $? != 0 ]; then
+		if [ -f $basedir/linux/tools/hv/lsvmbus ]; then
+			chmod +x $basedir/linux/tools/hv/lsvmbus
+			$basedir/linux/tools/hv/lsvmbus -vv > $lsvmbus_output_location
+		else
+			LogErr "File, lsvmbus, not found in the system. Aborted the execution."
+			SetTestStateAborted
+			exit 0
+		fi
+	else
+		lsvmbus -vv > $lsvmbus_output_location
+	fi
+	LogMsg "Successfully recorded lsvmbus output in $lsvmbus_output_location"
+
+	# Before chaning cpu id to 0, reset all cpu ids of each vmbus channel.
+	LogMsg "Resetting CPU ID if non-zero"
+	reset_cpu_id
+
+	# Select a CPU number where does not associate to vmbus channels from idle_cpus array
+	for id in ${idle_cpus[@]}; do
+		state=$(cat /sys/devices/system/cpu/cpu$id/online)
+		if [[ $state == "1" ]]; then
+			LogMsg "Verified the current cpu $id is online"
+			dmesg > /tmp/pre-stage.log
+			LogMsg "Took a snapshot of dmesg to /tmp/pre-stage.log"
+			# set cpu to offline
+			echo 0 > /sys/devices/system/cpu/cpu$id/online
+			LogMsg "Changed the cpu $id state to offline"
+			sleep 1
+			post_state=$(cat /sys/devices/system/cpu/cpu$id/online)
+			if [[ $post_state == "0" ]]; then
+				LogMsg "Successfully verified to change the cpu $id state to offline"
+				sleep 1
+				dmesg > /tmp/post-stage.log
+				diff_val=$(diff /tmp/pre-stage.log /tmp/post-stage.log)
+				if [[ $diff_val == *"CPU $id is now offline"* ]]; then
+					LogMsg "Successfully found dmesg log per cpu offline state change"
+					# Change back to online
+					echo 1 > /sys/devices/system/cpu/cpu$id/online
+					LogMsg "Changed the cpu $id state to online"
+					sleep 1
+					post_state=$(cat /sys/devices/system/cpu/cpu$id/online)
+					if [[ $post_state == "1" ]]; then
+						LogMsg "Successfully verified to change back the cpu $id state to online"
+						sleep 1
+						dmesg > /tmp/post-stage2.log
+						diff_val2=$(diff /tmp/post-stage.log /tmp/post-stage2.log)
+						if [[ $diff_val2 == *"smpboot: Booting Node"* ]]; then
+							LogMsg "Successfully found dmesg log per cpu online state change"
+						else
+							LogErr "Failed to verify cpu online state change. Expected smpboot: Booting Node, but found $diff_val2"
+						fi
+					else
+						LogErr "Failed to change back cpu $id to onlone"
```

Comment:
`LogErr "Failed to change back cpu $id to online"`

Reply:
fixed